dlp/src/main/java/dlp/snippets/InspectStringWithExclusionDict.java

```diff
@@ -0,0 +1,127 @@
+import com.google.privacy.dlp.v2.CustomInfoType.Dictionary;
+import com.google.privacy.dlp.v2.CustomInfoType.Dictionary.WordList;
+import com.google.privacy.dlp.v2.ExclusionRule;
+import com.google.privacy.dlp.v2.Finding;
+import com.google.privacy.dlp.v2.InfoType;
+import com.google.privacy.dlp.v2.InspectConfig;
+import com.google.privacy.dlp.v2.InspectContentRequest;
+import com.google.privacy.dlp.v2.InspectContentResponse;
+import com.google.privacy.dlp.v2.InspectionRule;
+import com.google.privacy.dlp.v2.InspectionRuleSet;
+import com.google.privacy.dlp.v2.MatchingType;
+import com.google.privacy.dlp.v2.ProjectName;
+import com.google.protobuf.ByteString;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class InspectStringWithExclusionDict {
+
+  // TODO: Delete me, only for testing
+  public static void main(String[] args) {
+    String projectId = "crwilson-dlp-experimental";
+    inspectStringWithExclusionDict(projectId, args[0], Arrays.asList(args[1]));
+  }
+
+  public static void inspectStringWithExclusionDict() {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "your-project-id";
+    String textToInspect = "Some email addresses: gary@example.com, example@example.com";
+    List<String> excludedMatchList = Arrays.asList("example@example.com");
+    inspectStringWithExclusionDict(projectId, textToInspect, excludedMatchList);
+  }
+
+  // Inspects the provided text, avoiding matches specified in the exclusion list.
+  public static void inspectStringWithExclusionDict(String projectId, String textToInspect,
+      List<String> excludedMatchList) {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the "close" method on the client to safely clean up any remaining background resources.
+    try (DlpServiceClient dlp = DlpServiceClient.create()) {
+      // Specify the project used for request.
+      ProjectName project = ProjectName.of(projectId);
+
+      // Specify the type and content to be inspected.
+      ByteContentItem byteItem =
+          ByteContentItem.newBuilder()
+              .setType(BytesType.TEXT_UTF8)
+              .setData(ByteString.copyFromUtf8(textToInspect))
+              .build();
+      ContentItem item = ContentItem.newBuilder().setByteItem(byteItem).build();
+
+      // Specify the type of info the inspection will look for.
+      List<InfoType> infoTypes = new ArrayList<>();
+      // See https://cloud.google.com/dlp/docs/infotypes-reference for complete list of info types.
+      for (String typeName : new String[]{"PHONE_NUMBER", "EMAIL_ADDRESS", "CREDIT_CARD_NUMBER"}) {
+        infoTypes.add(InfoType.newBuilder().setName(typeName).build());
+      }
+
+      // Construct an exclusion rule that will exclude matches from the specified excludedMatchList.
```

Comment:
Drop the repetition of the type, more effective to comment what the code is doing:

```
// Exclude matches from the specified excludedMatchList.
```

Reply:
Done